verify_rpms/rpm_verifier.py

```diff
@@ -233,6 +233,30 @@ def aggregate_results(processed_image_list: list[ProcessedImage]) -> dict[str, A
     return aggregated_results
 
 
+def generate_images_processed_result(
+    processed_images_list: list[ProcessedImage], image_url: str, image_digest: str
+) -> dict[str, Any]:
+    """
+    Generate a dictionary containing the list of digests of the processed images.
+    If the image reference provided is of an Image Manifest, this list will contain one digest.
+    If it was an Image Index, then the result will contain the digest of the Image Index and a
+    digest for each of the Image Manifests listed in the processed_images_list.
+    :param processed_images_list: a list pf processed images
+    :param image_url: Image URL of the image reference
+    :param image_digest: Image digest of the image reference
+    :return: Dictionary with the list of digests processed
+    """
+    images_processed_dict: dict[str, Any] = {
+        "image": {"pullspec": image_url, "digests": []}
+    }
+    # set unique values in case image_digest is also the image processed
+    digests_set = set(
+        [image_digest] + [image.image.split("@")[-1] for image in processed_images_list]
+    )
+    images_processed_dict["image"]["digests"] = list(digests_set)
```

Comment:
I think we can drop the initial deceleration and just keep this statement.
It will be less lines to maintain and I think it's also makes things clearer.
```suggestion
    images_processed_dict: dict[str, Any] = {
        "image": {"pullspec": image_url, "digests": list(digests_set)}
    }
```

Reply:
Can do even this and spare the additional long-name variable
```suggestion
    retrun {"image": {"pullspec": image_url, "digests": list(digests_set)}}
```

Reply:
Done